pkg/migration/controllers/migration.go

```diff
@@ -172,40 +193,43 @@ func (m *MigrationController) migrateVolumes(migration *storkv1.Migration) error
 		}
 	}
 
-	// Now check the status
-	volumeInfos, err := m.Driver.GetMigrationStatus(migration)
-	if err != nil {
-		return err
-	}
-	if volumeInfos == nil {
-		volumeInfos = make([]*storkv1.VolumeInfo, 0)
-	}
-	migration.Status.Volumes = volumeInfos
-	// Store the new status
-	err = sdk.Update(migration)
-	if err != nil {
-		return err
-	}
+	// Skip checking status if no volumes are being migrated
+	if len(migration.Status.Volumes) != 0 {
+		// Now check the status
+		volumeInfos, err := m.Driver.GetMigrationStatus(migration)
+		if err != nil {
+			return err
+		}
+		if volumeInfos == nil {
+			volumeInfos = make([]*storkv1.VolumeInfo, 0)
+		}
+		migration.Status.Volumes = volumeInfos
+		// Store the new status
+		err = sdk.Update(migration)
+		if err != nil {
+			return err
+		}
 
-	// Now check if there is any failure or success
-	// TODO: On failure of one volume cancel other migrations?
-	for _, vInfo := range volumeInfos {
-		// Return if we have any volume migrations still in progress
-		if vInfo.Status == storkv1.MigrationStatusInProgress {
-			logrus.Infof("Volume Migration still in progress: %v", migration.Name)
-			return nil
-		} else if vInfo.Status == storkv1.MigrationStatusFailed {
-			m.Recorder.Event(migration,
-				v1.EventTypeWarning,
-				string(vInfo.Status),
-				fmt.Sprintf("Error migrating volume %v: %v", vInfo.Volume, vInfo.Reason))
-			migration.Status.Stage = storkv1.MigrationStageFinal
-			migration.Status.Status = storkv1.MigrationStatusFailed
-		} else if vInfo.Status == storkv1.MigrationStatusSuccessful {
-			m.Recorder.Event(migration,
-				v1.EventTypeNormal,
-				string(vInfo.Status),
-				fmt.Sprintf("Volume %v migrated successfully", vInfo.Volume))
+		// Now check if there is any failure or success
+		// TODO: On failure of one volume cancel other migrations?
+		for _, vInfo := range volumeInfos {
+			// Return if we have any volume migrations still in progress
+			if vInfo.Status == storkv1.MigrationStatusInProgress {
+				logrus.Infof("Volume Migration still in progress: %v", migration.Name)
+				return nil
+			} else if vInfo.Status == storkv1.MigrationStatusFailed {
+				m.Recorder.Event(migration,
+					v1.EventTypeWarning,
+					string(vInfo.Status),
+					fmt.Sprintf("Error migrating volume %v: %v", vInfo.Volume, vInfo.Reason))
+				migration.Status.Stage = storkv1.MigrationStageFinal
+				migration.Status.Status = storkv1.MigrationStatusFailed
+			} else if vInfo.Status == storkv1.MigrationStatusSuccessful {
+				m.Recorder.Event(migration,
+					v1.EventTypeNormal,
+					string(vInfo.Status),
+					fmt.Sprintf("Volume %v migrated successfully", vInfo.Volume))
+			}
```

Comment:
we should collect all progress, failed and successful volumes and then dump events so that they are consistent.

The way it is now, we may or may not see `"Volume %v migrated successfully` events in the migration object if one of the volume is still in progress or failed and it comes before and after the successful volume.